src/main/io/gps/gps_config.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#ifdef USE_GPS
+
+#include <string.h>
+
+typedef enum {
+    UNDEF,
+    M5,
+    M6,
+    M7,
+    M8,
+    M9,
+    M10
+} ubloxVersion_e;
```

Comment:
Do we support M5? I am not sure.
```suggestion
typedef enum {
    UBX_VERSION_UNDEF = 0,
    UBX_VERSION_M5,
    UBX_VERSION_M6,
    UBX_VERSION_M7,
    UBX_VERSION_M8,
    UBX_VERSION_M9,
    UBX_VERSION_M10,
    UBX_VERSION_COUNT
} ubloxVersion_e;
```

Reply:
We only need to know the version if it affects how we talk to the GPS.  Afaik the main change where the old commands became deprecated was when they went past M8.  At this point functionally M9 = M10 I think. So we don't need to know if it's M9 or M10, just that it's above a certain firmware level where 'new' or 'different' commands are required.

Reply:
M10 understands all M8 commands, but the payload structure has changed, thats why we cant just comment out some parts or skip them. The order of some values is different. Thats why it didnt work before.

Reply:
I don't see a problem to have a "list" with all the different versions and detect the correct one. I think in this way the code is more "complete" for a future.

But for the GPS instructions, is better to use the basic "payload" for all and change it for bigger than M9, to let it rather simple, ignoring the model exactly.

Reply:
I have all the available protocol sheets, and have checked that our code should work for M5 and above.  I have no clue about before M5.  

If indeed it is flaky for M5, hopefully someone find that out in a not unsafe way.

I don't care what strings we use very much, but the longer ones will consume more flash space.  The `Mn` identifier refers to hardware version.  These identifies are only relevant to UBlox hardware and will only be used inside 'if UBlox' conditions, so we don't really need the `UBLOX` text in each name.

Perhaps `HW_UNDEF`... `HW_M10`?

Reply:
This comment has been implemented